agent/runner.go

```diff
@@ -47,10 +59,37 @@ func NewRunner(workEngine rpc.Peer, f rpc.Filter, h string, state *State, backen
 		hostname: h,
 		counter:  state,
 		backend:  backend,
+		backOff: backOff{
+			pollInitialInterval: backOffInit,
+			pollMaxInterval:     backOffMax,
+		},
 	}
 }
 
 func (r *Runner) Run(runnerCtx context.Context) error {
+	retry := backoff.NewExponentialBackOff()
```

Comment:
Why do you want to backoff exponential instead of linear here?

Reply:
Because if you fetched a task last time its more likely that there are some next to come

Else in the worst case wait 10s